values.yaml

```diff
@@ -25,6 +25,26 @@ ingress:
   # Name of tls secret to use on ingress
   tlsSecretName: ~
 
+  # Annotations always injected when configuring Ingress ojbect
+  annotations: |-
+      kubernetes.io/ingress.class: {{ .Values.ingress.class }}
```

Comment:
Given we have an explicit field for this, we should probably leave this one in the template, not in the space for "user" annotations.

Reply:
If we leave it in the template then the user is required to have a value for it. Are there cases where a user would not need this value?

Reply:
Yea, the `values.yaml` file does not have templating support since its the input to the actual templating. We may just need to make the corresponding Houston PR.

Reply:
@schnie I was able to do the templating via values.yaml.

```kind: Ingress
apiVersion: extensions/v1beta1
metadata:
  name: release-name-airflow-ingress
  labels:
    tier: airflow
    component: airflow-ingress
    release: release-name
    chart: "airflow-0.12.0-alpha.2"
    heritage: Tiller
  annotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "false"
    nginx.ingress.kubernetes.io/custom-http-errors: "403,404,502,503"
    nginx.ingress.kubernetes.io/auth-url: https://houston./v1/authorization
    nginx.ingress.kubernetes.io/auth-signin: https://app./login
    nginx.ingress.kubernetes.io/auth-response-headers: authorization, username, email
    nginx.ingress.kubernetes.io/proxy-cookie-path: / /release-name/
    nginx.ingress.kubernetes.io/configuration-snippet: |
      if ($host = 'release-name-airflow.' ) {
        rewrite ^ https://deployments.%!s(<nil>)/release-name/airflow permanent;
      }
      if ($host = 'release-name-flower.' ) {
        rewrite ^ https://deployments.%!s(<nil>)/release-name/flower permanent;
      }```

This was generated when enabled=true

Reply:
I just had to use the tpl function https://github.com/helm/helm/issues/2133#issuecomment-589695803

Reply:
If anything I think these example values offer pretty good documentation.

Reply:
That output looks a little off. For example the `auth-url` is missing the `baseDomain` part and there are some `<nil>`s in the snippets. Is it because that value isn't set or because of a templating issue?

Reply:
@schnie I'm pretty certain it's just because the value isn't set. I'm just using the 'values.yaml' in the repo and not adding anything.

Reply:
```
kind: Ingress
apiVersion: extensions/v1beta1
metadata:
  name: release-name-airflow-ingress
  labels:
    tier: airflow
    component: airflow-ingress
    release: release-name
    chart: "airflow-0.12.0-alpha.2"
    heritage: Tiller
  annotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "false"
    nginx.ingress.kubernetes.io/custom-http-errors: "403,404,502,503"
    nginx.ingress.kubernetes.io/auth-url: https://houston.foobar.com/v1/authorization
    nginx.ingress.kubernetes.io/auth-signin: https://app.foobar.com/login
    nginx.ingress.kubernetes.io/auth-response-headers: authorization, username, email
    nginx.ingress.kubernetes.io/proxy-cookie-path: / /release-name/
    nginx.ingress.kubernetes.io/configuration-snippet: |
      if ($host = 'release-name-airflow.foobar.com' ) {
        rewrite ^ https://deployments.foobar.com/release-name/airflow permanent;
      }
      if ($host = 'release-name-flower.foobar.com' ) {
        rewrite ^ https://deployments.foobar.com/release-name/flower permanent;
      }
spec:
```

This is what it looks like when I added a basedomain

Reply:
ping @schnie is this concern resolved?